src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -382,6 +394,21 @@ private async Task ProcessQueueCoreAsync<TRequest, TResponse>(
         }
     }
 
+    private string TryGetLanguageForRequest(IQueueItem<TRequestContext> work,
+        out Exception? ex)
+    {
+        try
+        {
+            var language = _languageServer.GetLanguageForRequest(work.MethodName, work.SerializedRequest);
```

Comment:
can we make the server just detect this case (possibly with an optional parameter controlling if it throws or not), so we only do this for some requests, not all.